PennMobile/src/main/java/com/pennapps/labs/pennmobile/LaundryFragment.java

```diff
@@ -0,0 +1,216 @@
+package com.pennapps.labs.pennmobile;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.ListFragment;
+import android.view.LayoutInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.ListView;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import com.pennapps.labs.pennmobile.adapters.LaundryHallAdapter;
+import com.pennapps.labs.pennmobile.api.Labs;
+import com.pennapps.labs.pennmobile.classes.Laundry;
+import com.pennapps.labs.pennmobile.classes.LaundryHall;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import butterknife.Bind;
+import butterknife.ButterKnife;
+import rx.functions.Action1;
+
+
+public class LaundryFragment extends ListFragment {
+
+    private Labs mLabs;
+    private ListView mListView;
+    private MainActivity mActivity;
+    private final static int ROW_CAP = 15;
+    @Bind(R.id.loadingPanel) RelativeLayout loadingPanel;
+    @Bind(R.id.no_results) TextView no_results;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        mLabs = MainActivity.getLabsInstance();
+        mActivity = (MainActivity) getActivity();
+        mActivity.closeKeyboard();
+    }
+
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        setHasOptionsMenu(true);
+        mListView = getListView();
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View v = inflater.inflate(R.layout.fragment_laundry, container, false);
+        ButterKnife.bind(this, v);
+
+        getLaundryHall();
+        return v;
+    }
+
+    private void getLaundryHall() {
+        mLabs.laundries()
+                .subscribe(new Action1<List<Laundry>>() {
+                    @Override
+                    public void call(final List<Laundry> laundryHalls) {
```

Comment:
it's confusing to name a list of Laundries "laundryHalls", if LaundryHall is a different type of object